.github/workflows/bleeding-edge.yaml

```diff
@@ -19,33 +22,39 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    name: Python3.10-dev
+    name: "tests with bleeding-edge crucial deps"
     timeout-minutes: 60
 
     concurrency:
-      group: ${{ github.ref }}-dev
+      # auto-cancel any in-progress job *on the same branch*
+      group: ${{ github.workflow }}-${{ github.ref }}
       cancel-in-progress: true
 
     steps:
-    - uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
+    - name: Checkout repo
+      uses: actions/checkout@v3
 
-    - name: Set up Python Dev Version
-      uses: actions/setup-python@v2
+    - name: Set up Python (newest testable version)
+      uses: actions/setup-python@v3
       with:
+        # the '-dev' suffix allows to use alphas and betas if no final release is available yet
+        # this version should be upgraded as often as possible, typically once a year when numpy
+        # and Cython are known to be compatible
+        # for next update (to '3.11-dev', 2022), see https://github.com/numpy/numpy/pull/21308
         python-version: '3.10-dev'
 
     - name: Install dependencies
+      # regarding why pip 22.1 is forbidden, see https://github.com/pypa/pip/issues/11116
+      # this can be simplified when a newer version is released
       run: |
-        python -m pip install --upgrade pip
+        python -m pip install --upgrade "pip != 22.1"
```

Comment:
Should we create an issue to track this?

Reply:
I don't think it's needed. It should work forever ™️  and can be cleaned at any point when the condition is met.

Reply:
pip 22.1.1 is out. This patch is now longer needed, I removed it